metis-normalization/src/main/java/eu/europeana/normalization/dates/extraction/dateextractors/DcmiPeriodDateExtractor.java

```diff
@@ -43,27 +46,26 @@ public class DcmiPeriodDateExtractor implements DateExtractor {
       "name" + EQUALS_SPACES_WRAPPED + "([^;]*|[^$]*)" + VALUE_ENDING);
 
   private static final Set<String> W3C_DTF_SCHEME_VALUES = Set.of("W3C-DTF", "W3CDTF");
-  private static final EdtfParser EDTF_PARSER = new EdtfParser();
 
   @Override
-  public DateNormalizationResult extract(String value) {
-    DateNormalizationResult dateNormalizationResult = null;
+  public DateNormalizationResult extract(String value, DateQualification requestedDateQualification,
+      boolean flexibleDateBuild) throws DateExtractionException {
+    DateNormalizationResult dateNormalizationResult = DateNormalizationResult.getNoMatchResult(value);
     if (isValidScheme(value)) {
-      try {
-        Matcher matcher = DCMI_PERIOD_START_PATTERN.matcher(value);
-        InstantEdtfDate start = extractDate(matcher);
-        matcher = DCMI_PERIOD_END_PATTERN.matcher(value);
-        InstantEdtfDate end = extractDate(matcher);
-        String name = extractName(value);
+      Matcher matcher = DCMI_PERIOD_START_PATTERN.matcher(value);
+      InstantEdtfDate start = extractDate(matcher, requestedDateQualification, flexibleDateBuild);
+      matcher = DCMI_PERIOD_END_PATTERN.matcher(value);
+      InstantEdtfDate end = extractDate(matcher, requestedDateQualification, flexibleDateBuild);
+      String name = extractName(value);
 
-        //At least one end has to be specified
-        if (!start.getEdtfDatePart().isUnspecified() || !end.getEdtfDatePart().isUnspecified()) {
-          IntervalEdtfDate intervalEdtfDate = new IntervalEdtfDate(name, start, end);
-          dateNormalizationResult = new DateNormalizationResult(DateNormalizationExtractorMatchId.DCMI_PERIOD, value,
-              intervalEdtfDate);
-        }
-      } catch (DuplicateFieldException | ParseException e) {
-        LOGGER.warn("Exception during dcmi field extraction", e);
+      //At least one end has to be specified
+      if (start.getDateEdgeType() == DateEdgeType.DECLARED || end.getDateEdgeType() == DateEdgeType.DECLARED) {
```

Comment:
the method was not renamed :sweat_smile: `start.getDateEdgeType()` => `start.getDateBoundaryType()`

Reply:
aah the IDE cheated on me.. :D